### DISCUSSION TITLE
Error connecting MCP to sqlite server: mcp-server-sqlite version

### DISCUSSION THREAD

**Discussion by qu-genesis** [modelcontextprotocol/.github#18] [Q&A] [answered]
Hello! Super excited about this, I'm running into what seems like a silly error hoping that folks here can help:

I ran the code in the Quick Start guide and having been getting what seems to be a Python version error:
```
==> /Users/qugenes/Library/Logs/Claude/mcp-server-sqlite.log <==
  × No solution found when resolving tool dependencies:
  ╰─▶ Because the current Python version (3.9.6) does not satisfy Python>=3.10
      and mcp-server-sqlite<=0.4.0 depends on Python>=3.11, we can conclude
      that mcp-server-sqlite<=0.4.0 cannot be used.
      And because only the following versions of mcp-server-sqlite are
      available:
          mcp-server-sqlite==0.1.0
          mcp-server-sqlite==0.3.0
          mcp-server-sqlite==0.3.1
          mcp-server-sqlite==0.4.0
          mcp-server-sqlite==0.4.1
          mcp-server-sqlite==0.5.0
          mcp-server-sqlite==0.5.1
      we can conclude that mcp-server-sqlite<0.4.1 cannot be used. (1)

      Because the current Python version (3.9.6) does not satisfy Python>=3.10
      and mcp-server-sqlite>=0.4.1 depends on Python>=3.10, we can conclude
      that mcp-server-sqlite>=0.4.1 cannot be used.
      And because we know from (1) that mcp-server-sqlite<0.4.1 cannot be
      used, we can conclude that all versions of mcp-server-sqlite cannot
      be used.
      And because you require mcp-server-sqlite, we can conclude that your
      requirements are unsatisfiable.
```
Here are my version specifications, I'm assuming the MCP is trying to run my system's builtin python instead of the miniconda python, but I'm at a loss how to switch them. Would love a bit of guidance, thank you!

```
➜  ~ which python
/Users/qugenes/miniconda3/bin/python
➜  ~ python --version
Python 3.11.9
➜  ~ /usr/bin/python3 --version                            
Python 3.9.6
➜  ~ uv --version
uv 0.5.4 (c62c83c37 2024-11-20)

```

**Comment by chizhikchi** [comment]
Hi! 
I'm having similar issues in my M1 MacBook Air: the MCP is trying to un my system's Python 3.8 (which should not be updated following Apple's guidelines). 
Will be grateful for any suggestions on how this error can be overcome.

**Reply by dsp-ant** [reply]
Hmm. This is probably a better question for the `uv` folks on how the platform selection actually works. One idea would be to use `uvx --python 3.11` to force `uv` into selecting and potentially downloading, or use `uvx --python-preference=only-managed`.

**Reply by chizhikchi** [reply, marked as answer]
Thanks for your reply! 
 I was able to fix this issue by specifying the path to python executable like following: 
 ```
      "sqlite": {
        "command": "uvx",
        "args": ["--python", "PATH_TO_PYTHON", "mcp-server-sqlite", "--db-path", "/Users/user/test.db"]
      }
```

**Reply by qu-genesis** [reply]
Me too thank you both for the help!
